vernac/synterp.ml

```diff
@@ -409,9 +409,19 @@ let with_succeed f =
   Vernacstate.Synterp.unfreeze st;
   ControlSucceed { st = transient_st } :: ctrl, v
 
+let synpure_control : control_flag -> control_entry =
```

Comment:
Wouldn't it be better to share the code with `synterp_control_flag` ?

Reply:
not that easy, synterp_control_flag needs to handle the case with Fail where the command fails in synterp phase so the interp phase is turned into a noop

Not worth the effort imo

Reply:
I agree with you in terms of effort, let's stay like this for now indeed.

But the whole API setup is not good, we should really classify the commands with types.